sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -283,3 +590,970 @@ def get_lpmode_support(self):
+            
+            if ImageBValid == 0:
+                # Registers 9Fh:174,175; 176.177
+                ImageB = '%d.%d.%d' %(rpl[38], rpl[39], ((rpl[40]<< 8) | rpl[41]))
+            else:
+                ImageB = "N/A"
+            logger.info('Image B Version: %s' %ImageB)
+
+            if ImageARunning == 1:
+                RunningImage = 'A'
+            elif ImageBRunning == 1:
+                RunningImage = 'B'
+            if ImageACommitted == 1:
+                CommittedImage = 'A'
+            elif ImageBCommitted == 1:
+                CommittedImage = 'B'
+            logger.info('Running Image: %s; Committed Image: %s' %(RunningImage, CommittedImage))
+        else:
+            raise ValueError('Reply payload check code error')
+        elapsedtime = time.time()-starttime
+        logger.info('Get module FW info time: %.2f s' %elapsedtime)
+        return ImageA, ImageARunning, ImageACommitted, ImageAValid, ImageB, ImageBRunning, ImageBCommitted, ImageBValid
+
+    def module_fw_run(self, mode = 0x01):
+        """
+        This command is used to start and run a selected image.
+        This command transfers control from the currently
+        running firmware to a selected firmware that is started. It
+        can be used to switch between firmware versions, or to
+        perform a restart of the currently running firmware.
+        mode:
+        00h = Traffic affecting Reset to Inactive Image.
+        01h = Attempt Hitless Reset to Inactive Image
+        02h = Traffic affecting Reset to Running Image.
+        03h = Attempt Hitless Reset to Running Image
+        """
+        try:
+            self.cdb
+        except AttributeError:
+            self.get_cdb_api()
+        # run module FW (CMD 0109h)
+        starttime = time.time()
+        fw_run_status = self.cdb.run_fw_image(mode)
+        if fw_run_status == 1:
+            logger.info('Module FW run: Success')
+        else:
+            self.cdb.abort_fw_download()
+            logger.info('Module FW run: Fail')
+        elapsedtime = time.time()-starttime
+        logger.info('Module FW run time: %.2f s\n' %elapsedtime)
+
+    def module_fw_commit(self):
+        """
+        The host uses this command to commit the running image
+        so that the module will boot from it on future boots.
+        """
+        try:
+            self.cdb
+        except AttributeError:
+            self.get_cdb_api()
+        # commit module FW (CMD 010Ah)
+        starttime = time.time()
+        fw_commit_status= self.cdb.commit_fw_image()
+        if fw_commit_status == 1:
+            logger.info('Module FW commit: Success')
+        else:
+            self.cdb.abort_fw_download()
+            logger.info('Module FW commit: Fail')
+        elapsedtime = time.time()-starttime
+        logger.info('Module FW commit time: %.2f s\n' %elapsedtime)
+
+    def module_fw_download(self, startLPLsize, maxblocksize, lplonly_flag, autopaging_flag, writelength, imagepath):
```

Comment:
please add return status for failure/success

Reply:
done